yb-voyager/src/tgtdb/event.go

```diff
@@ -258,9 +258,14 @@ func (eb EventBatch) GetLastVsn() int64 {
 	return eb.Events[len(eb.Events)-1].Vsn
 }
 
-func (eb EventBatch) GetQueryToUpdateLastAppliedVSN(migrationUUID uuid.UUID) string {
-	return fmt.Sprintf(`UPDATE %s SET last_applied_vsn=%d where migration_uuid='%s' AND channel_no=%d`,
-		EVENT_CHANNELS_METADATA_TABLE_NAME, eb.GetLastVsn(), migrationUUID, eb.ChanNo)
+func (eb EventBatch) GetQueryToUpdateLastAppliedVSN(migrationUUID uuid.UUID, numInserts int64, numUpdates int64, numDeletes int64) string {
```

Comment:
Almost ALWAYS the receiver should be a pointer i.e. in this case it should be `func (eb *EventBatch)`. Otherwise, the value will be passed by-value which can be costly depending on the object.

As a general guideline: consider the event-receiver as a normal parameter passed to the function.